tsnet/tsnet_test.go

```diff
@@ -867,16 +874,94 @@ func TestUserMetrics(t *testing.T) {
 	want := `# TYPE tailscaled_advertised_routes gauge
 # HELP tailscaled_advertised_routes Number of advertised network routes (e.g. by a subnet router)
 tailscaled_advertised_routes 2
+# TYPE tailscaled_approved_routes gauge
+# HELP tailscaled_approved_routes Number of approved network routes (e.g. by a subnet router)
+tailscaled_approved_routes 1
 # TYPE tailscaled_health_messages gauge
 # HELP tailscaled_health_messages Number of health messages broken down by type.
 tailscaled_health_messages{type="warning"} 2
 # TYPE tailscaled_inbound_dropped_packets_total counter
 # HELP tailscaled_inbound_dropped_packets_total Counts the number of dropped packets received by the node from other peers
 # TYPE tailscaled_outbound_dropped_packets_total counter
 # HELP tailscaled_outbound_dropped_packets_total Counts the number of packets dropped while being sent to other peers
+# TYPE tailscaled_primary_routes gauge
+# HELP tailscaled_primary_routes Number of primary network routes (e.g. by a subnet router)
+tailscaled_primary_routes 1
 `
 
 	if diff := cmp.Diff(want, string(metrics1)); diff != "" {
 		t.Fatalf("unexpected metrics (-want +got):\n%s", diff)
 	}
 }
+
+func TestUserMetricsNetwork(t *testing.T) {
```

Comment:
This test is currently just used for local testing, I think it should be folded into the other one when I find a good way to separate the metrics from the two tsnets.